helix-core/src/lib.rs

```diff
@@ -92,6 +92,14 @@ pub fn config_dir() -> std::path::PathBuf {
     path
 }
 
+/// Searches for the local `.helix` directory by searching for the root `.git` directory,
+/// using the CWD if it can't find one.
+pub fn local_config_dir() -> std::path::PathBuf {
+    let root = find_root(None)
+        .unwrap_or_else(|| std::env::current_dir().expect("unable to determine current directory"));
+    root.join(".helix")
+}
```

Comment:
Shouldn't we check if the cwd contains a .helix directory first ? So in effect:

1. Explicitly check if cwd has `.helix`. If so use that. If not...
2. Search for the root dir. If found use `root/.helix`. If not...
3. Assume `cwd/.helix` (which would not exist).

I remember someone opening an issue about having a .git repo in their home dir and helix calculating the wrong root.